src/inference/include/openvino/runtime/intel_gpu/remote_properties.hpp

```diff
@@ -185,5 +185,34 @@ static constexpr Property<uint32_t> dev_object_handle{"DEV_OBJECT_HANDLE"};
  */
 static constexpr Property<uint32_t> va_plane{"VA_PLANE"};
 
+// TODO: maybe find more suitable place for these properties (should be accessible from both
+// src/plugins/intel_gpu/.../mlir_op.cpp and src/common/transformations/.../mlir_op.cpp)
+/**
+ * @brief Namespace for properties related to MLIR operations within the GPU plugin.
+ * These properties are used as evaluation context parameters for MLIR operations,
+ * assisting in managing events, result tracking, and kernel argument types.
+ */
+namespace mlir_meta {
+
+/**
+ * @brief This key identifies an event list to wait for a kernel execution.
+ * @ingroup ov_runtime_ocl_gpu_cpp_api
+ */
+static constexpr Property<std::vector<gpu_handle_param>> wait_list{"EVENTS_WAIT_LIST"};
+
+/**
+ * @brief This key identifies a pointer to a cl::Event that should be set with
+ * the result cl_event of a kernel execution.
+ * @ingroup ov_runtime_ocl_gpu_cpp_api
+ */
+static constexpr Property<gpu_handle_param> result_event{"RESULT_EVENT"};
+
+/**
+ * @brief This key identifies whether the kernel argument at [i] position is USM pointer
+ * @ingroup ov_runtime_ocl_gpu_prop_cpp_api
+ */
+static constexpr Property<std::vector<bool>> is_kernel_arg_usm{"IS_KERNEL_ARG_USM"};
```

Comment:
This header is a part of user API. So I think these properties need to be moved to internal header. `src/inference/dev_api/openvino/runtime/internal_properties.hpp` could be an option